### DISCUSSION TITLE
How to practice java code

### DISCUSSION THREAD

**Discussion by Sahil-SDE** [codewars/codewars.com#2824] [Q&A]
Suggest to me how to practice java code and enhance my coding skills

**Comment by z-b-b** [comment]
Hi, there are java exercises (including solutions) through this website >>
https://www.w3resource.com/java-exercises/

**Comment by ghost** [comment]
You can practice your Java skills by using websites such as codewars.com, leetcode.com, and hackerrank.com. Read books, do small projects, attend meetups, and watch YouTube videos when you need clarification on topics. 

Here’s a great book for Java beginners: https://beginnersbook.com/java-tutorial-for-beginners-with-examples/. Also, try building small projects.

Here’s a great link for Java project ideas: https://www.geeksforgeeks.org/top-50-java-project-ideas-for-beginners-advanced/amp/.
